packages/platforms/celestica/x86-64/x86-64-cel-redstone-xp/onlp/builds/src/module/src/fani.c

```diff
@@ -0,0 +1,111 @@
+#include <onlp/platformi/fani.h>
+
+#include "i2c_chips.h"
+#include "platform.h"
+
+onlp_fan_info_t f_info[FAN_COUNT + 1] = {
+    {
+        { ONLP_FAN_ID_CREATE(1), "Chassis Fan 1", 0 },
+        0x0,
+    },
+    {
+        { ONLP_FAN_ID_CREATE(2), "Chassis Fan 2", 0 },
+        0x0,
+    },
+    {
+        { ONLP_FAN_ID_CREATE(3), "Chassis Fan 3", 0 },
+        0x0,
+    },
+    {
+        { ONLP_FAN_ID_CREATE(4), "Chassis Fan 4", 0 },
+        0x0,
+    },
+    {
+        { ONLP_FAN_ID_CREATE(5), "Chassis Fan 5", 0 },
+        0x0,
+    },
+    {
+        { ONLP_FAN_ID_CREATE(6), "Chassis Fan 6", 0 },
+        0x0,
+    },
+    {
+        { ONLP_FAN_ID_CREATE(7), "Chassis Fan 7", 0 },
+        0x0,
+    },
+    {
+        { ONLP_FAN_ID_CREATE(8), "Chassis Fan 8", 0 },
+        0x0,
+    },
+    {
+        { ONLP_FAN_ID_CREATE(8), "PSU Fan 1", 0 },
+        0x0,
+    },
+    {
+        { ONLP_FAN_ID_CREATE(8), "PSU Fan 2", 0 },
+        0x0,
+    },
+};
+
+
+int
+onlp_fani_init(void)
+{
+    fanInit();
+    return ONLP_STATUS_OK;
+}
+
+int
+onlp_fani_rpm_set(onlp_oid_t id, int rpm)
+{
+    int fan_id;
+    unsigned short p;
+
+    /* Max speed 12000 RPM. 1 % is 120 RPM */
```

Comment:
The MAX RPM value is usually encoded in the YAML file and then used as a define. 
For example:
https://github.com/opencomputeproject/OpenNetworkLinux/blob/master/packages/platforms/quanta/powerpc/powerpc-quanta-ly2/onlp/builds/src/module/auto/powerpc_quanta_ly2.yml#L35